### PR TITLE
[WIP} Fix type annotation in pandas/core/api

### LINKED ISSUE #26148: Fix Type Annotation in pandas/core/api.py
Part of #25882 

Current errors are:

    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'Int8Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'Int16Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'Int32Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'Int64Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'UInt8Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'UInt16Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'UInt32Dtype'
    pandas\core\api.py:5: error: Module 'pandas.core.arrays.integer' has no attribute 'UInt64Dtype'

### PR DESCRIPTION
**Pull request by vaibhavhrt**
- [x] closes #26148 
- [x] tests passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by vaibhavhrt** [comment]
@WillAyd please take a look. Mypy is showing a lot of errors in `setup.py` and one in `pandas\core\computation\expr.py` when running locally, not sure why.

File: pandas/core/arrays/integer.py [hunk 1]
```diff
@@ -696,22 +696,77 @@ def integer_arithmetic_method(self, other):
...
+    type('UInt16Dtype', (_IntegerDtype, ), {
+        'type': getattr(np, 'uint16'),
+        'name': 'UInt16',
+        '__doc__': _dtype_docstring.format(dtype='uint16')
+    })
+)
+
+UInt32Dtype = register_extension_dtype(
+    type('UInt32Dtype', (_IntegerDtype, ), {
+        'type': getattr(np, 'uint8'),
```
Comment (WillAyd) [inline pandas/core/arrays/integer.py]: Typo here I think is causing test failures

File: pandas/core/arrays/integer.py [hunk 2]
```diff
@@ -696,22 +696,77 @@ def integer_arithmetic_method(self, other):
...
-                       '__doc__': _dtype_docstring.format(dtype=dtype)}
-    dtype_type = register_extension_dtype(
-        type(classname, (_IntegerDtype, ), attributes_dict)
-    )
-    setattr(module, classname, dtype_type)
-
-    _dtypes[dtype] = dtype_type()
+Int8Dtype = register_extension_dtype(
+    type('Int8Dtype', (_IntegerDtype, ), {
+        'type': getattr(np, 'int8'),
```
Comment (WillAyd) [inline pandas/core/arrays/integer.py]: No need to use getattr for these just access normally

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks! Some minor issues as commented. Didn’t see the Mypy issue you mentioned above so maybe not an issue

File: pandas/core/arrays/integer.py (hunk 1 above)
Comment (vaibhavhrt) [reply pandas/core/arrays/integer.py]: ah yeah i see. thanks for figuring it out.

File: pandas/core/arrays/integer.py (hunk 2 above)
Comment (vaibhavhrt) [reply pandas/core/arrays/integer.py]: just `np.int8` ?

**Comment by vaibhavhrt** [comment]
@WillAyd I have addressed all the issues and tests should pass now. But codecov says coverage decreasing by 51%, do I need to write some tests too?

**Comment by jreback** [comment]
this is fine. waiting on the resolution of the CI, then will have you rebase.

**Comment by jreback** [comment]
thanks @vaibhavhrt

**Comment by vaibhavhrt** [comment]
Part of #25882

**Pull request merged**